agent/src/control_interface/control_interface_task.rs

```diff
@@ -97,10 +97,20 @@ impl ControlInterfaceTask {
 
     pub async fn run(mut self) {
         // [impl->swdd~agent-closes-control-interface-on-missing-initial-hello~1]
-        if let Err(message) = self.check_initial_hello().await {
-            log::warn!("{message}");
-            let _ = self.send_connection_closed(message).await;
-            return;
+        match self.check_initial_hello().await {
+            Ok(_) => {
+                log::info!("Control interface connection established.");
+                // [impl->swdd~agent-sends-accepted-message-on-valid-initial-hello~1]
+                if let Err(err) = self.send_control_interface_accepted().await {
+                    log::warn!("Could not send ControlInterfaceAccepted message: '{err}'");
+                    return;
+                }
```

Comment:
Just a suggestion to pull the info log down.
```suggestion
                // [impl->swdd~agent-sends-accepted-message-on-valid-initial-hello~1]
                if let Err(err) = self.send_control_interface_accepted().await {
                    log::warn!("Could not send ControlInterfaceAccepted message: '{err}'");
                    return;
                }
                log::info!("Control interface connection established.");
```